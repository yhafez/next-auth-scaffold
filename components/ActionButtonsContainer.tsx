// Path: ./components/ActionButtonsContainer.tsx
import { ReactNode } from 'react'
import { Box } from '@mui/material'

export interface ActionButtonsContainerProps {
	name: string
	children: ReactNode
}

export default function ActionButtonsContainer({ name, children }: ActionButtonsContainerProps) {
	return (
		<Box
			id={`${name}-buttons-container`}
			sx={{
				display: 'flex',
				flexDirection: 'row',
				justifyContent: 'space-around',
				alignItems: 'center',
				gap: 2,
				margin: '8px 4px',
				width: '80%',
			}}
		>
			{children}
		</Box>
	)
}
