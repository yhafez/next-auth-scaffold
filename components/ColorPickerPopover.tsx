// Path: ./components/ColorPickerPopover.tsx

import { useState, useCallback } from 'react'
import { Popover, PopoverOrigin } from '@mui/material'
import { useBoundStore } from '../store'
import { ChromePicker } from 'react-color'

export interface ColorPickerPopoverProps {
	name: string
	open: boolean
	anchorEl: HTMLButtonElement | null
	handleClose: () => void
	handleColorChange: (color: string) => void
	anchorOrigin?: PopoverOrigin
	transformOrigin?: PopoverOrigin
}

const ColorPickerPopover = ({
	name,
	open,
	anchorEl,
	handleClose,
	handleColorChange,
	anchorOrigin,
	transformOrigin,
}: ColorPickerPopoverProps) => {
	const { customPalette } = useBoundStore()

	const [color, setColor] = useState(customPalette.primary.main)

	const handleChange = useCallback(
		(color: any) => {
			setColor(color.hex)
			handleColorChange(color.hex)
		},
		[handleColorChange],
	)

	return (
		<Popover
			id={`${name}-color-picker-popover`}
			open={open}
			anchorEl={anchorEl}
			onClose={handleClose}
			anchorOrigin={anchorOrigin}
			transformOrigin={transformOrigin}
			sx={{
				cursor: 'pointer',
				'& .MuiPopover-paper': {
					padding: 2,
					backgroundColor: 'background.paper',
					color: 'text.primary',
				},
			}}
		>
			<ChromePicker color={color} onChange={color => handleChange(color)} disableAlpha />
		</Popover>
	)
}

export default ColorPickerPopover
