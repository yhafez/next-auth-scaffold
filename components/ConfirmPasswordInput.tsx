// Path: ./components/ConfirmPasswordInput.tsx
import { Dispatch, SetStateAction, KeyboardEvent } from 'react'

import PasswordInput from './PasswordInput'

export interface ConfirmPasswordInputProps {
	name: string
	password: string
	setPassword: Dispatch<SetStateAction<string>>
	confirmPassword: string
	setConfirmPassword: Dispatch<SetStateAction<string>>
	disabled?: boolean
	handleEnter?: (e: KeyboardEvent) => void
}

export default function ConfirmPasswordInput({
	name,
	password,
	setPassword,
	confirmPassword,
	setConfirmPassword,
	disabled = false,
	handleEnter,
}: ConfirmPasswordInputProps) {
	return (
		<>
			<PasswordInput
				name={name}
				value={password}
				setValue={setPassword}
				disabled={disabled}
				handleEnter={handleEnter}
			/>
			<PasswordInput
				name={`${name}`}
				value={confirmPassword}
				setValue={setConfirmPassword}
				disabled={disabled}
				isConfirmPassword={true}
				handleEnter={handleEnter}
			/>
		</>
	)
}
