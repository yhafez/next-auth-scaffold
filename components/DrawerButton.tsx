// Path: ./components/DrawerButton.tsx
import { ReactNode, MouseEvent } from 'react'
import Link from 'next/link'
import { useRouter } from 'next/router'
import { Link as MuiLink, IconButton, Typography } from '@mui/material'

export interface DrawerButtonProps {
	name: string
	label: string
	icon: ReactNode
	handleClick?: (e: MouseEvent<HTMLButtonElement>) => void
	reverse?: boolean
	href?: string
}

export default function DrawerButton({
	name,
	label,
	icon,
	handleClick,
	reverse,
	href,
}: DrawerButtonProps) {
	const router = useRouter()
	const isSelected = router.pathname === href ? true : false

	if (href) {
		return (
			<MuiLink
				id={`${name}-drawer-link`}
				component={Link}
				href={href}
				aria-current={isSelected ? 'page' : undefined}
				title={label}
				aria-labelledby={`${name}-drawer-button-label`}
				sx={{
					textDecoration: 'none',
					color: 'primary.contrastText',
					padding: 0,
					display: 'flex',
					alignItems: 'center',
					justifyContent: 'space-between',
					width: '100%',
					borderRadius: 0,
					backgroundColor: isSelected ? 'primary.main' : 'transparent',
					p: 4,
					'&:hover': {
						backgroundColor: 'primary.main',
					},
					'&:focus': {
						backgroundColor: 'primary.main',
					},
					'&:active': {
						backgroundColor: 'primary.main',
					},
				}}
			>
				{reverse && icon}
				<Typography
					id={`${name}-drawer-button-label`}
					variant="body1"
					sx={{
						fontSize: '1.2rem',
						fontWeight: 'bold',
					}}
				>
					{label}
				</Typography>
				{!reverse && icon}
			</MuiLink>
		)
	}

	return (
		<IconButton
			id={`${name}-drawer-button`}
			onClick={handleClick}
			disableRipple
			sx={{
				color: 'primary.contrastText',
				padding: 0,
				display: 'flex',
				alignItems: 'center',
				justifyContent: 'space-between',
				width: '100%',
				borderRadius: 0,
				backgroundColor: isSelected ? 'primary.main' : 'transparent',
				p: 4,
				'&:hover': {
					backgroundColor: 'primary.main',
				},
				'&:focus': {
					backgroundColor: 'primary.main',
				},
				'&:active': {
					backgroundColor: 'primary.main',
				},
			}}
		>
			{reverse && icon}
			<Typography
				id={`${name}-drawer-button-label`}
				variant="body1"
				sx={{
					fontSize: '1.2rem',
					fontWeight: 'bold',
				}}
			>
				{label}
			</Typography>
			{!reverse && icon}
		</IconButton>
	)
}
