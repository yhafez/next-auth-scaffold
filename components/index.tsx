// Path: ./components/index.tsx
import Layout from './Layout'

import DashboardDrawer from './Drawers/DashboardDrawer'
import SettingsDrawer from './Drawers/SettingsDrawer'

import ColorPickerPopover from './Popovers/ColorPickerPopover'
import DrawerSignOutPopover from './Popovers/DrawerSignOutPopover'

import ActionButtonsContainer from './ActionButtonsContainer'
import ColorPickerIcon from './ColorPickerIcon'
import ConfirmPasswordInput from './ConfirmPasswordInput'
import DarkModeSwitch from './DarkModeSwitch'
import Drawer from './Drawer'
import DrawerButton from './DrawerButton'
import EditableTextField from './EditableTextField'
import EmailInput from './EmailInput'
import Modal from './Modal'
import ModalButton from './ModalButton'
import ModalNote from './ModalNote'
import NavigationButton from './NavigationButton'
import PasswordInput from './PasswordInput'
import SubmitButton from './SubmitButton'
import VisibilityIconButton from './VisibilityIconButton'

export {
	Layout,
	DashboardDrawer,
	SettingsDrawer,
	ColorPickerPopover,
	DrawerSignOutPopover,
	ActionButtonsContainer,
	ColorPickerIcon,
	ConfirmPasswordInput,
	DarkModeSwitch,
	Drawer,
	DrawerButton,
	EditableTextField,
	EmailInput,
	Modal,
	ModalButton,
	ModalNote,
	NavigationButton,
	PasswordInput,
	SubmitButton,
	VisibilityIconButton,
}
