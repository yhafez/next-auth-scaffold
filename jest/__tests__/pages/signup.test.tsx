// Path: ./jest/__tests__/pages/signup.test.tsx
import { render, act, screen, waitFor } from '@testing-library/react'
import { axe } from 'jest-axe'
import { SessionProvider } from 'next-auth/react'
import { ThemeProvider } from '@mui/material/styles'

import Signup from '../../../pages/signup'
import { Layout } from '../../../components'
import { defaultTheme } from '../../../theme'

describe('Signup', () => {
	let container: HTMLElement
	beforeEach(() => {
		act(() => {
			container = render(
				<SessionProvider
					session={{
						token: {
							user: {
								id: '1',
								email: 'test@example.com',
								emailVerified: null,
								image: 'test',
								isDeactivated: false,
								name: null,
								role: 'USER',
								createdAt: new Date(),
								updatedAt: new Date(),
								isNewUser: false,
							},
						},
						expires: '1d',
					}}
				>
					<ThemeProvider theme={defaultTheme}>
						<Layout name="test" pageTitle="test">
							<Signup />
						</Layout>
					</ThemeProvider>
				</SessionProvider>,
			).container

			return container
		})
	})

	it('Should have no accessibility violations', async () => {
		waitFor(async () => expect(await axe(screen.getByRole('main'))).toHaveNoViolations())
	})

	it('Should match snapshot', () => {
		expect(container).toMatchSnapshot()
	})
})
