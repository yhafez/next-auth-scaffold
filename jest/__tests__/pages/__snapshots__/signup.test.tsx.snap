// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14umlvy-MuiButtonBase-root-MuiButton-root"
    id="test-skip-nav-link"
    tabindex="0"
    type="button"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-3so7zb-MuiTypography-root"
      id="test-skip-nav-link-text"
    >
      Skip to main content
    </p>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <div
    class="MuiBox-root css-114gqq5"
    id="test-layout-container"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-38ln4h-MuiPaper-root-MuiAppBar-root"
      id="test-app-bar"
    >
      <nav
        aria-label="header navigation"
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1l1du5z-MuiToolbar-root"
        id="test-toolbar"
      >
        <a
          aria-label="Home"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fqdykp-MuiTypography-root-MuiLink-root"
          href="/"
          id="test-toolbar-title-link"
          title="Home"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 MuiTypography-noWrap css-63r4vv-MuiTypography-root"
            id="test-toolbar-title"
          >
            Next Auth Scaffold
          </h1>
        </a>
        <ul
          class="MuiList-root MuiList-padding css-166iuh2-MuiList-root"
          id="test-toolbar-icons"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-16ztiij-MuiListItem-root"
            id="test-toolbar-icons-color-picker-item"
          >
            <div
              class="MuiBox-root css-1lo2k45"
              id="test-toolbar-color-picker-container"
            >
              <button
                aria-labelledby="test-toolbar-color-picker-label"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                id="test-toolbar-color-picker-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  aria-labelledby="test-toolbar-color-picker-label"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
                  data-testid="ColorLensIcon"
                  focusable="false"
                  id="test-toolbar-color-picker-icon"
                  role="img"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <label
                for="test-toolbar-color-picker-button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-7xs8rl-MuiTypography-root"
                  id="test-toolbar-color-picker-label"
                >
                  Theme Color
                </p>
              </label>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9gmft3-MuiListItem-root"
            id="test-toolbar-icons-dark-mode-switch-item"
          >
            <div
              class="MuiBox-root css-f9wnac"
              id="test-toolbar-dark-mode-switch-container"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-41alfi-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-28iv8m-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    aria-checked="false"
                    aria-labelledby="test-toolbar-dark-mode-switch-label-light"
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    id="test-toolbar-dark-mode-switch"
                    role="switch"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="false"
                    aria-labelledby="test-toolbar-dark-mode-switch-label-light"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq4ggj-MuiSvgIcon-root"
                    data-testid="WbSunnyIcon"
                    focusable="false"
                    id="test-toolbar-dark-mode-switch-icon-unchecked"
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <label
                for="test-toolbar-dark-mode-switch"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-191g3cx-MuiTypography-root"
                  id="test-toolbar-dark-mode-switch-label-light"
                >
                  Light
                  <br />
                   Mode
                </p>
              </label>
            </div>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="MuiBox-root css-cn70eu"
      id="test-layout-content"
      tabindex="-1"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14umlvy-MuiButtonBase-root-MuiButton-root"
        id="signup-skip-nav-link"
        tabindex="0"
        type="button"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-3so7zb-MuiTypography-root"
          id="signup-skip-nav-link-text"
        >
          Skip to main content
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiBox-root css-114gqq5"
        id="signup-layout-container"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-38ln4h-MuiPaper-root-MuiAppBar-root"
          id="signup-app-bar"
        >
          <nav
            aria-label="header navigation"
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1l1du5z-MuiToolbar-root"
            id="signup-toolbar"
          >
            <a
              aria-label="Home"
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fqdykp-MuiTypography-root-MuiLink-root"
              href="/"
              id="signup-toolbar-title-link"
              title="Home"
            >
              <h1
                class="MuiTypography-root MuiTypography-h1 MuiTypography-noWrap css-63r4vv-MuiTypography-root"
                id="signup-toolbar-title"
              >
                Next Auth Scaffold
              </h1>
            </a>
            <ul
              class="MuiList-root MuiList-padding css-166iuh2-MuiList-root"
              id="signup-toolbar-icons"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-16ztiij-MuiListItem-root"
                id="signup-toolbar-icons-color-picker-item"
              >
                <div
                  class="MuiBox-root css-1lo2k45"
                  id="signup-toolbar-color-picker-container"
                >
                  <button
                    aria-labelledby="signup-toolbar-color-picker-label"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    id="signup-toolbar-color-picker-button"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-labelledby="signup-toolbar-color-picker-label"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
                      data-testid="ColorLensIcon"
                      focusable="false"
                      id="signup-toolbar-color-picker-icon"
                      role="img"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <label
                    for="signup-toolbar-color-picker-button"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-7xs8rl-MuiTypography-root"
                      id="signup-toolbar-color-picker-label"
                    >
                      Theme Color
                    </p>
                  </label>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9gmft3-MuiListItem-root"
                id="signup-toolbar-icons-dark-mode-switch-item"
              >
                <div
                  class="MuiBox-root css-f9wnac"
                  id="signup-toolbar-dark-mode-switch-container"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-41alfi-MuiSwitch-root"
                  >
                    <span
                      class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-28iv8m-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        aria-checked="false"
                        aria-labelledby="signup-toolbar-dark-mode-switch-label-light"
                        class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                        id="signup-toolbar-dark-mode-switch"
                        role="switch"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="false"
                        aria-labelledby="signup-toolbar-dark-mode-switch-label-light"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq4ggj-MuiSvgIcon-root"
                        data-testid="WbSunnyIcon"
                        focusable="false"
                        id="signup-toolbar-dark-mode-switch-icon-unchecked"
                        role="img"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <label
                    for="signup-toolbar-dark-mode-switch"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-191g3cx-MuiTypography-root"
                      id="signup-toolbar-dark-mode-switch-label-light"
                    >
                      Light
                      <br />
                       Mode
                    </p>
                  </label>
                </div>
              </li>
            </ul>
          </nav>
        </header>
        <main
          class="MuiBox-root css-cn70eu"
          id="signup-layout-content"
          tabindex="-1"
        >
          <div
            class="MuiBox-root css-1bghj2m"
            id="signup-modal-container"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-118nrrt-MuiTypography-root"
              id="signup-modal-title-word-0"
            >
              Signup
            </p>
            <div
              class="MuiBox-root css-7n00rz"
              id="signup-modal-notification-container"
            />
            <form
              class="MuiBox-root css-xo10vp"
              id="signup-modal-form"
              role="form"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-8v3hey-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-7yavjh-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="signup-email-input"
                  id="signup-email-input-label"
                >
                  Email
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-8r7mze-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="signup-email-input"
                    type="email"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Email
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root css-8v3hey-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-7yavjh-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="signup-password-input"
                  id="signup-password-input-label"
                >
                  Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-7kjzy7-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="signup-password-input"
                    type="password"
                    value=""
                  />
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-dpfpu4-MuiButtonBase-root-MuiIconButton-root"
                    id="signup-password-input-show-hide-button"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="MuiBox-root css-erqh4x"
                      id="signup-password-box"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="show password"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
                        data-testid="VisibilityOffIcon"
                        focusable="false"
                        id="signup-password-input-visibility-off-icon"
                        role="button"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                        />
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption css-5oi5yi-MuiTypography-root"
                        id="signup-password-input-show-password-text"
                      >
                        Show
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root css-8v3hey-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-7yavjh-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="signup-confirm-password-input"
                  id="signup-confirm-password-input-label"
                >
                  Confirm Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-7kjzy7-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="signup-confirm-password-input"
                    type="password"
                    value=""
                  />
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-dpfpu4-MuiButtonBase-root-MuiIconButton-root"
                    id="signup-confirm-password-input-show-hide-button"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="MuiBox-root css-erqh4x"
                      id="signup-confirm-password-box"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="show password"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
                        data-testid="VisibilityOffIcon"
                        focusable="false"
                        id="signup-confirm-password-input-visibility-off-icon"
                        role="button"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                        />
                      </svg>
                      <span
                        class="MuiTypography-root MuiTypography-caption css-5oi5yi-MuiTypography-root"
                        id="signup-confirm-password-input-show-password-text"
                      >
                        Show
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Confirm Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiBox-root css-ueupuo"
                id="signup-buttons-container"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1bu79mp-MuiButtonBase-root-MuiButton-root"
                  id="signup-submit-button"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTypography-root MuiTypography-button css-11gc47n-MuiTypography-root"
                    id="signup-submit-button-text"
                  >
                    Sign Up
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiBox-root css-1h5af1d"
                id="signup-login-modal-note-container"
              >
                <a
                  aria-label="Already have an account? Log in"
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ptl0h6-MuiTypography-root-MuiLink-root"
                  href="/login"
                  id="signup-login-modal-note-link"
                  title="Already have an account? Log in"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1qwu16v-MuiTypography-root"
                    id="signup-login-modal-note-link-text"
                  >
                    Already have an account? Log in
                  </p>
                </a>
              </div>
            </form>
          </div>
        </main>
      </div>
    </main>
  </div>
</div>
`;
